internal/konnect/client.go

```diff
@@ -1,39 +1,43 @@
 package konnect
 
 import (
-	"bytes"
-	"context"
 	"crypto/tls"
-	"encoding/json"
 	"fmt"
-	"io"
 	"net/http"
-	neturl "net/url"
-	"strconv"
 
 	"github.com/kong/kubernetes-ingress-controller/v2/internal/adminapi"
 	tlsutil "github.com/kong/kubernetes-ingress-controller/v2/internal/util/tls"
 )
 
-// NodeAPIClient is used for sending requests to Konnect Node API.
-// It can be used to register Nodes in Konnect's Runtime Groups.
-type NodeAPIClient struct {
+// Client is a Client for the Konnect API, scoped to a single runtime group.
+type Client struct {
 	Address        string
 	RuntimeGroupID string
 	Client         *http.Client
+	Common         konnectResourceClient
+	Node           AbstractNodeAPI
+	License        AbstractLicenseAPI
```

Comment:
```suggestion
	Node           *NodeAPIClient
	License       *LicenseAPIClient
```
These two don't have to be interface types as `Client` is keeping a concrete implementation.

Reply:
Since discussion of using the go-kong model has indicated we actually don't like it and want something else, I've reverted the original refactor and instead put in a simpler solution that just mirrors the existing node agent implementation: the clients are instantiated separately and have some code duplication as a result.

I think we do need to unify them, but time constraints (I turn into a pumpkin for a month next week) aren't going to allow for designing a proper alternative, so I think we need a stopgap option with some debt for now.

Reply:
Works for me 👍